crates/timings/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+use std::{collections::VecDeque, sync::atomic::AtomicBool, time::Duration};
+
+use ambient_ecs::{components, Debuggable, FrameEvent, Resource, System, World, WorldContext};
+use ambient_sys::time::Instant;
+use num_derive::{FromPrimitive, ToPrimitive};
+use num_traits::FromPrimitive;
+use winit::event::{DeviceEvent, Event, WindowEvent};
+
+const MAX_SAMPLES: usize = 128;
+static ENABLED: AtomicBool = AtomicBool::new(false);
+
+pub fn set_enabled(enabled: bool) {
+    ENABLED.store(enabled, std::sync::atomic::Ordering::Relaxed);
+}
+
+/// Frame events being timed (in order!)
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, FromPrimitive, ToPrimitive)]
+pub enum TimingEventType {
```

Comment:
Would it be possible to make this less hardcoded and instead use a `&'static str` instead?

Right now, this creates a conceptual dependency between all of ambients system and our internals, and this sub-crate, which increases coupling even more.

Let me know if I am being too hard on the review, have been in the open-source reviewing sphere lately where they are *very* strict 😅

Reply:
That sounds good, I suppose this will be a case of seeing which part goes the easier, like when playing Jenga.

I'd personally go with the core approach, as adding another crate for it may not be worthwhile over the increased link time and risks of cyclic dependencies.